libcontainer/factory_linux.go

```diff
@@ -274,32 +274,25 @@ func validateID(id string) error {
 	return nil
 }
 
-func parseMountFds() ([]int, error) {
-	fdsJSON := os.Getenv("_LIBCONTAINER_MOUNT_FDS")
+func parsEnvFds(envVariable string) ([]int, error) {
+	fdsJSON := os.Getenv(envVariable)
 	if fdsJSON == "" {
 		// Always return the nil slice if no fd is present.
 		return nil, nil
 	}
 
 	var mountFds []int
 	if err := json.Unmarshal([]byte(fdsJSON), &mountFds); err != nil {
-		return nil, fmt.Errorf("Error unmarshalling _LIBCONTAINER_MOUNT_FDS: %w", err)
+		return nil, fmt.Errorf("Error unmarshalling %s: %w", envVariable, err)
 	}
 
 	return mountFds, nil
 }
 
-func parseIdmapFds() ([]int, error) {
-	fdsJSON := os.Getenv("_LIBCONTAINER_IDMAP_FDS")
-	if fdsJSON == "" {
-		// Always return the nil slice if no fd is present.
-		return nil, nil
-	}
-
-	var mountFds []int
-	if err := json.Unmarshal([]byte(fdsJSON), &mountFds); err != nil {
-		return nil, fmt.Errorf("Error unmarshalling _LIBCONTAINER_IDMAP_FDS: %w", err)
-	}
+func parseMountFds() ([]int, error) {
+	return parsEnvFds("_LIBCONTAINER_MOUNT_FDS")
+}
 
-	return mountFds, nil
+func parseIdmapFds() ([]int, error) {
```

Comment:
I think not having two different functions for this, and just call `parseEnvFds()` is way simpler